app/platform/config_test.go

```diff
@@ -98,3 +100,84 @@ func TestConfigSave(t *testing.T) {
 		metricsMock.AssertNumberOfCalls(t, "Register", 1)
 	})
 }
+
+func TestIsFirstUserAccount(t *testing.T) {
+	th := SetupWithStoreMock(t)
+	defer th.TearDown()
+	storeMock := th.Service.Store.(*smocks.Store)
+	userStoreMock := &smocks.UserStore{}
+	storeMock.On("User").Return(userStoreMock)
+
+	t.Run("success no users", func(t *testing.T) {
+		*userStoreMock = smocks.UserStore{}
+
+		userStoreMock.On("Count", model.UserCountOptions{IncludeDeleted: true}).Return(int64(0), nil)
+		require.True(t, th.Service.IsFirstUserAccount())
+	})
+
+	t.Run("success one user", func(t *testing.T) {
+		*userStoreMock = smocks.UserStore{}
+
+		userStoreMock.On("Count", model.UserCountOptions{IncludeDeleted: true}).Return(int64(1), nil)
+		require.False(t, th.Service.IsFirstUserAccount())
+	})
+
+	t.Run("success multiple users", func(t *testing.T) {
+		*userStoreMock = smocks.UserStore{}
+
+		userStoreMock.On("Count", model.UserCountOptions{IncludeDeleted: true}).Return(int64(42), nil)
+		require.False(t, th.Service.IsFirstUserAccount())
+	})
+
+	t.Run("success negative users", func(t *testing.T) {
+		*userStoreMock = smocks.UserStore{}
+
+		userStoreMock.On("Count", model.UserCountOptions{IncludeDeleted: true}).Return(int64(-100), nil)
+		require.True(t, th.Service.IsFirstUserAccount())
+	})
+
+	t.Run("failed request", func(t *testing.T) {
+		*userStoreMock = smocks.UserStore{}
+
+		userStoreMock.On("Count", model.UserCountOptions{IncludeDeleted: true}).Return(int64(0), errors.New("error"))
+		require.False(t, th.Service.IsFirstUserAccount())
+	})
+
+	// create a session, this should not affect IsFirstUserAccount
+	th.Service.sessionCache.Set("mock_session", 1)
+
+	t.Run("success no users with session", func(t *testing.T) {
```

Comment:
Not a big deal, it seems we are just repeating the same batch of tests. I'd probably run 2 parent loops - one with session, one without session, and with these sub-tests. 0/5

Reply:
Yeah, good point. This makes adding additional tests easier.